lib/fumanchu/compiler.ex

```diff
@@ -3,11 +3,49 @@ defmodule FuManchu.Compiler do
   alias FuManchu.Parser
   alias FuManchu.Generator
 
+  @spec compile(String.t) :: (map -> String.t)
```

Comment:
Hmm. Not entirely sure the spec, in either form, is accurate. Can compile return an error tuple or raise an error?